apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -260,4 +271,24 @@ export class SendMessageInApp extends SendMessageType {
       })
     );
   }
+
+  private async processAvatar(
+    avatarDetails: IAvatarDetails,
+    environmentId: string,
+    actorId?: string
+  ): Promise<string | null> {
+    if (avatarDetails.type === AvatarTypeEnum.USER && actorId) {
+      const actorSubscriber: SubscriberEntity = await this.subscriberRepository.findOne(
+        {
+          _environmentId: environmentId,
+          _id: actorId,
+        },
+        'avatar'
+      );
+
+      return actorSubscriber?.avatar || null;
+    }
+
+    return avatarDetails.data || null;
+  }
```

Comment:
A thought. Should a failure inside this method (when requesting the avatar of a subscriber) break all the flow of sending an inApp message? As it will bubble up.
Personally I do not consider it should and we should handle it with a try/catch plus logging/reporting to Sentry to be able to identify potential bugs.
What do you think?

Reply:
If i understood correctly, we should wrap the `processAvatar` with a `try/catch`, report the error to sentry, and inside the `catch` instead of throwing error, `return null` as fallback value so that the notification flow keeps moving?

Reply:
Yes, exactly that. For this use case _hiding_ the error would be a good thing as would improve the UX.

Reply:
ok understood, thanks will add it